examples/pendulum_swing_up_variable_duration.py

```diff
@@ -0,0 +1,120 @@
+"""This solves the simple pendulum swing up problem presented here:
+
+http://hmc.csuohio.edu/resources/human-motion-seminar-jan-23-2014
+
+A simple pendulum is controlled by a torque at its joint. The goal is to swing
+the pendulum from its rest equilibrium to a target angle by minimizing the
+energy used to do so in a minimal amount of time.
+
+Solves the same problem as in ``pendulum_swing_up.py`` but with a variable
+total duration.
+
+"""
+
+from collections import OrderedDict
+
+import numpy as np
+import sympy as sym
+from opty.direct_collocation import Problem
+from opty.utils import building_docs
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+
+target_angle = np.pi
+num_nodes = 500
+save_animation = False
+
+# Symbolic equations of motion
+m, g, d, t, h = sym.symbols('m, g, d, t, h')
+theta, omega, T = sym.symbols('theta, omega, T', cls=sym.Function)
+
+state_symbols = (theta(t), omega(t))
+constant_symbols = (m, g, d)
+specified_symbols = (T(t),)
+
+eom = sym.Matrix([theta(t).diff() - omega(t),
+                  m*d**2*omega(t).diff() + m*g*d*sym.sin(theta(t)) - T(t)])
+
+# Specify the known system parameters.
+par_map = OrderedDict()
+par_map[m] = 1.0
+par_map[g] = 9.81
+par_map[d] = 1.0
+
+
+# Specify the objective function and it's gradient.
+def obj(free):
+    """Minimize the sum of the squares of the control torque."""
+    T = free[2 * num_nodes:]
+    return free[-1]*np.sum(T**2)
+
+
+def obj_grad(free):
+    T = free[2 * num_nodes:]
+    grad = np.zeros_like(free)
+    grad[2 * num_nodes:] = 2.0 * free[-1] * free[2 * num_nodes:]
+    grad[-1] = np.sum(T**2)
+    return grad
+
+
+# Specify the symbolic instance constraints, i.e. initial and end conditions
+# using node numbers 1 to N.
+instance_constraints = (theta(1),
+                        theta(num_nodes) - target_angle,
+                        omega(1),
+                        omega(num_nodes))
```

Comment:
When providing the node index here instead of time, we could ask for node numbers 0  to N - 1 or 1 to N.

I am a bit unsure which choice is the better. I guess it doesn't matter as long as it is clearly stated in the documentation.

Reply:
When referring to index I would say that from 0 is more logical.